baseplate/metrics.py

```diff
@@ -85,22 +85,62 @@ def send(self, serialized_metric):
 
 class BufferedTransport(object):
     """A transport which wraps another transport and buffers before sending."""
+    MAX_MESSAGE_SIZE = 65507
+
     def __init__(self, transport):
         self.transport = transport
         self.buffer = []
 
     def send(self, serialized_metric):
-        self.buffer.append(serialized_metric)
+        if len(serialized_metric) >= self.MAX_MESSAGE_SIZE:
+            logger.warning("skipping too-long metric")
+        else:
+            self.buffer.append(serialized_metric)
+
+    def make_messages(self, metrics):
+        if not metrics:
+            return []
+
+        num_newlines = len(metrics) - 1
+        num_chars = sum(len(metric) for metric in metrics)
+        need_batching = num_newlines + num_chars > self.MAX_MESSAGE_SIZE
+        if not need_batching:
+            return [b"\n".join(metrics)]
+
+        messages = []
+        batch = []
+        batch_len = 0
+        for metric in metrics:
+            if batch:
+                proposed_len = batch_len + len(metric) + 1  # newline + metric
+            else:
+                proposed_len = len(metric)
+
+            if proposed_len > self.MAX_MESSAGE_SIZE:
+                messages.append(b"\n".join(batch))
+                batch = []
+                batch_len = 0
+
+            batch_len += len(metric)
+            if batch:
+                batch_len += 1  # count the newline
```

Comment:
It might be good to detail this just a touch more, it took me some thinking to realize that this is counting the newline between the previous value in `batch` and the new one, _not_ the newline that comes after the current metric (I thought you were double counting the newline when checking proposed_len for a min).